big_tests/tests/mod_filter_SUITE.erl

```diff
@@ -0,0 +1,156 @@
+%       can only chat with themselves. so that particular host
+%       will have no connection to the exterior.
+%    2. anonymous users of some host (anonymous.localhost)
+%       can only chat with shared single chat user(some bot, i.e kate@localhost), 
+%       so that particular host will have no connection
+%       to the other hosts (localhost, sogndal) listed in hosts 
+update_rules()->
+    {atomic,ok} = rpc(mim(), acl, add, [global, domain1, {server_glob, <<"localhost">>}]),
+    {atomic,ok} = rpc(mim(), acl, add, [global, domain2, {server_glob, <<"sogndal">>}]),
+    {atomic,ok} = rpc(mim(), acl, add, [global, bot, {user, <<"kate">>, <<"localhost">>}]),
+    {atomic,ok} = rpc(mim(), acl, add, [global, anon_user, {server_glob, <<"anonymous.localhost">>}]),
+
+
+    {atomic,ok} = rpc(mim(), ejabberd_config, add_global_option, [{access, mod_filter, global}, [
+                                                                                        {allow, all}]]),
+    {atomic,ok} = rpc(mim(), ejabberd_config, add_global_option, [{access, mod_filter_presence, global},
+                                                                                        [{allow, all}]]),
+    {atomic,ok} = rpc(mim(), ejabberd_config, add_global_option, [{access, mod_filter_message, global}, [
+                                                                                        {local, bot},
+                                                                                        {restrict_bot, anon_user},
+                                                                                        {restrict_dom1, domain1},
+                                                                                        {restrict_dom2, domain2}
+                                                                                        ]]),
+    {atomic,ok} = rpc(mim(), ejabberd_config, add_global_option, [{access, mod_filter_iq, global}, 
+                                                                                        [{allow, all}]]),
+
+    %restriction for localhost
+    {atomic,ok} = rpc(mim(), ejabberd_config, add_global_option, [{access, restrict_dom1, global}, 
+                                                                  [{allow, domain1}, {deny, all}]]),
+    %restriction for sogndal
+    {atomic,ok} = rpc(mim(), ejabberd_config, add_global_option, [{access, restrict_dom2, global},
+                                                                  [{allow, domain2}, {deny, all}]]),
+    %restriction for bot
+    {atomic,ok} = rpc(mim(), ejabberd_config, add_global_option, [{access, restrict_bot, global}, 
+                                                                  [{allow, bot}, {deny, all}]]).
+%%--------------------------------------------------------------------
+%% Message tests
+%%--------------------------------------------------------------------
+
+allowed_messages_story(Config) ->
```

Comment:
Here we also have code with a common part. Let's create a function like `user_can_send_to(From, To)`